src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -11,68 +11,224 @@
 
 namespace Symfony\Component\Cache\Adapter;
 
+use Symfony\Component\Cache\Exception\CacheException;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
 /**
  * @author Rob Frawley 2nd <rmf@src.run>
+ * @author Nicolas Grekas <p@tchwork.com>
  */
 class MemcachedAdapter extends AbstractAdapter
 {
+    private static $defaultClientOptions = array(
+        'persistent_id' => null,
+        'username' => null,
+        'password' => null,
+        'binary_protocol' => true,
+        'libketama_compatible' => true,
+    );
+
+    protected $maxIdLength = 250;
+
     private $client;
 
+    public static function isSupported()
+    {
+        return extension_loaded('memcached') && version_compare(phpversion('memcached'), '2.2.0', '>=');
+    }
+
     public function __construct(\Memcached $client, $namespace = '', $defaultLifetime = 0)
     {
+        if (!static::isSupported()) {
+            throw new CacheException('Memcached >= 2.2.0 is required');
+        }
+        $opt = $client->getOption(\Memcached::OPT_SERIALIZER);
+        if (\Memcached::SERIALIZER_PHP !== $opt && \Memcached::SERIALIZER_IGBINARY !== $opt) {
+            throw new CacheException('MemcachedAdapter: serializer must be PHP or IGBINARY.');
+        }
+        if (!$client->getOption(\Memcached::OPT_BINARY_PROTOCOL)) {
+            throw new CacheException('MemcachedAdapter: OPT_BINARY_PROTOCOL must be used.');
+        }
+        if ($client->getOption(\Memcached::OPT_NO_BLOCK)) {
+            throw new CacheException('MemcachedAdapter: OPT_NO_BLOCK must be disabled.');
+        }
```

Comment:
Regarding `binary_protocol` requirement: I'm on board with this.

But, regarding `no_block` requirement: I disagree. The intended use of this option doesn't require any additional code, and, in fact, writing such code would invalidate the entire intended usage and benefits of `no_block`, anyway. The expectation when it is enabled is that the user doesn't need *guarantees* when reading/writing. It is a **commonly used option** and generally doesn't change expectations; due to the design of memcached, one can never assume a value exists as it may have been purged at any time by the LRU algorithm, and will definitely not exist after a restart of the daemon.